meta-sokol-flex-common/classes/cb-mbs-options.bbclass

```diff
@@ -0,0 +1,139 @@
+                    _, check_value = check_flag.split('=', 1)
+                    if check_flag in cflags:
+                        cflags.remove(check_flag)
+                    if setting and not d.getVarFlag('CB_MBS_OPTIONS', skey, True):
+                        d.setVarFlag('CB_MBS_OPTIONS', skey, svalue + check_value)
+
+            for check_flag in check_cxxflags:
+                if check_flag.startswith(flag + '='):
+                    _, check_value = check_flag.split('=', 1)
+                    if check_flag in cxxflags:
+                        cxxflags.remove(check_flag)
+                    if setting and not d.getVarFlag('CB_MBS_OPTIONS', skey, True):
+                        d.setVarFlag('CB_MBS_OPTIONS', skey, svalue + check_value)
+
+    for flag, settings in (d.getVarFlags('CB_MBS_OPTIONS_CC_FLAGS_MAP') or {}).items():
+        for setting in settings.split():
+            if flag in check_cflags:
+                if flag in cflags:
+                    cflags.remove(flag)
+                if setting:
+                    skey, svalue = setting.split('=', 1)
+                    skey = 'gnu.c.' + skey
+                    svalue = 'gnu.c.' + svalue
+                    if not d.getVarFlag('CB_MBS_OPTIONS', skey, True):
+                        d.setVarFlag('CB_MBS_OPTIONS', skey, svalue)
+
+            if flag in check_cxxflags:
+                if flag in cxxflags:
+                    cxxflags.remove(flag)
+                if setting:
+                    skey, svalue = setting.split('=', 1)
+                    skey = 'gnu.cpp.' + skey
+                    svalue = 'gnu.cpp.compiler.' + svalue
+                    if not d.getVarFlag('CB_MBS_OPTIONS', skey, True):
+                        d.setVarFlag('CB_MBS_OPTIONS', skey, svalue)
+
+    ignored = d.getVar('CB_MBS_IGNORED_EXTRA_FLAGS', True).split()
+    cflags = [a for a in cflags if not any(fnmatchcase(a, p) for p in ignored)]
+    cxxflags = [a for a in cxxflags if not any(fnmatchcase(a, p) for p in ignored)]
+    ldflags = [a for a in ldflags if not any(fnmatchcase(a, p) for p in ignored)]
+
+    if cflags:
+        d.setVarFlag('CB_MBS_OPTIONS', 'gnu.c.compiler*option.misc.other', subprocess.list2cmdline(cflags))
+    if cxxflags:
+        d.setVarFlag('CB_MBS_OPTIONS', 'gnu.cpp.compiler*option.other.other', subprocess.list2cmdline(cxxflags))
+    if ldflags:
+        d.setVarFlag('CB_MBS_OPTIONS', 'gnu.c.link*option.ldflags', subprocess.list2cmdline(ldflags))
+        d.setVarFlag('CB_MBS_OPTIONS', 'gnu.cpp.link*option.flags', subprocess.list2cmdline(ldflags))
+
+    for option, value in d.getVarFlags('CB_MBS_OPTIONS').items():
+        value = d.expand(value)
+        if not value:
+            d.delVarFlag('CB_MBS_OPTIONS', option)
+        else:
+            # Force immediate expansion, so we use target overrides regardless
+            # of the context in which the flags are used
+            d.setVarFlag('CB_MBS_OPTIONS', option, value)
```

Comment:
this class is relocated from meta-mentor-private, right?

Reply:
Yep

Reply:
And I think it would make sense to identify in the commit message where (repo/branch/hash) we picked these up from.

Reply:
The commits now mention where the files came from, if not their revisions due to their being modified before pulling them over.